File: Serene1/Serene1/Serene1.Web/Modules/Common/Imports/ServerTypings/Northwind.ShipperForm.ts
﻿namespace Serene1.Northwind {
    export class ShipperForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Shipper';

    }

    export interface ShipperForm {
        CompanyName(): Serenity.StringEditor;
        Phone(): PhoneEditor;
    }

    [['CompanyName', Serenity.StringEditor], ['Phone', PhoneEditor]].forEach(x => ShipperForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

File: Serene1/Serene1/Serene1.Script/Imports/ServerImports/Northwind.ShipperForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene1.Northwind
{
    public partial class ShipperForm : PrefixedContext
    {
        [InlineConstant]
        public const string FormKey = "Northwind.Shipper";

        public ShipperForm(string idPrefix) : base(idPrefix) { }

        public StringEditor CompanyName { [InlineCode("{this}.w('CompanyName', Serenity.StringEditor)")] get; private set; }
        public PhoneEditor Phone { [InlineCode("{this}.w('Phone', Serene1.Northwind.PhoneEditor)")] get; private set; }
    }
}

